src/os/bluestore/BlueStore.h

```diff
@@ -513,8 +513,9 @@ class BlueStore : public ObjectStore,
       ++nref;
     }
     void put() {
-      if (--nref == 0)
-	delete this;
+      if (--nref == 0) {
+	//delete this;
+      }
```

Comment:
Another concern (applicable to both pools) - we return a 'dirty' object, i.e one that contains some information from the prev round - IMO this might cause some issues as post-allocation code might rely on some specific (clean) object state...